src/Aquifer.Jobs/Subscribers/ResourceContentVersionSimilarityMessageSubscriber.cs

```diff
@@ -0,0 +1,239 @@
+        int machineTranslationId,
+        AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        return await dbContext
+                .ResourceContentVersionMachineTranslations
+                .AsTracking()
+                .FirstOrDefaultAsync(m => m.Id == machineTranslationId, ct)
+            ?? throw new InvalidOperationException($"Machine translation with id {machineTranslationId} not found");
+    }
+    
+    private static async Task<ResourceContentVersionMachineTranslationEntity> GetMachineTranslationForResourceContentVersion(
+        int versionId, 
+        AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        return await dbContext
+                   .ResourceContentVersionMachineTranslations
+                   .AsTracking()
+                   .Where(x => x.ResourceContentVersionId == versionId)
+                   .OrderBy(x => x.Id)
+                   .LastOrDefaultAsync(ct) 
+               ?? throw new InvalidOperationException( $"No Machine translation for ResourceContentVersion with id {versionId} found");
+    }
+    
+    private static async Task<ResourceContentVersionMachineTranslationEntity> GetMachineTranslationForSnapshot(
+        int versionId, 
+        AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        return await dbContext
+                       .ResourceContentVersionMachineTranslations
+                       .AsTracking()
+                       .Join(
+                           dbContext.ResourceContentVersionSnapshots,
+                           rcvmt => rcvmt.ResourceContentVersionId,
+                           rcvs => rcvs.ResourceContentVersionId,
+                           (rcvmt, rcvs) => new {rcvmt, rcvs}
+                       )
+                       .Where(joined => joined.rcvs.Id == versionId)
+                       .Select(x => x.rcvmt)
+                       .OrderBy(x => x.Id)  
+                       .LastOrDefaultAsync(ct) 
+                   ?? throw new InvalidOperationException(
+                       $"No Machine translation for ResourceContentVersionSnapshot with id {versionId} found"
+                   );
+    }
+    
+    private static async Task<string> GetBaseVersionContentText(
+        int resourceContentVersionId,
+        AquiferDbContext dbContext,
+        CancellationToken ct)
+    {
+        var resource = await dbContext
+                           .ResourceContentVersions
+                           .AsTracking()
+                           .SingleOrDefaultAsync(x => x.Id == resourceContentVersionId, ct)
+                       ?? throw new InvalidOperationException($"Content version with id {resourceContentVersionId} not found");
+        
+        var resourceContentHtmlItems = TiptapConverter.ConvertJsonToHtmlItems(resource.Content);
+        
+        return HtmlUtilities.GetPlainText(
+            string.Join(string.Empty, resourceContentHtmlItems)
+        );
```

Comment:
C# styling generally prefers parentheses to not be on their own line and instead be at the end of the preceding line.  I very rarely make exclusions to this rule but usually only for tuples where the parentheses are acting more like braces.

All that said, I'd just put this on a single line (but see my question below on if we should even be doing this string join):
```suggestion
        return HtmlUtilities.GetPlainText(string.Join(string.Empty, resourceContentHtmlItems));
```